src/api/posts.ts

```diff
@@ -0,0 +1,25 @@
+import { Post } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
+  const response = await fetch(`${BASE_URL}/posts?userId=${userId}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+}
+
+export async function getPostbyId(postId:number) {
+  const response = await fetch(`${BASE_URL}/posts/${postId}`);
+
+  return response.json();
+}
+
+export async function getAllPosts() {
```

Comment:
1. Add return types to the functions for better type safety Promise<Post>.
2. add try catch handler:

```
export async function getAllPosts(): Promise<Post[]> {
  try {
    const response = await fetch(`${BASE_URL}/posts`);

    if (!response.ok) {
      throw new Error(`${response.status} - ${response.statusText}`);
    }

    return response.json();
  } catch (error) {
    console.error('Failed to fetch posts:', error);
    throw error;
  }
}
```